### PR TITLE
Fixes #207, supplying AWS_DEFAULT_REGION fixes v0.12 `tf validate`

### LINKED ISSUE #207: `make terraform/validate` does not to run on Terraform v0.12.X
## Describe the Bug
Running `make terraform/validate` on Terraform v0.12.X prints the following message: 
> Terraform 0.12 does not support validate without skipping variables

## Expected Behavior
I would expect the `terraform/validate` target to not care about this `skipping variables` flag as it doesn't seem needed, but maybe I am missing something.

## Steps to Reproduce
1. `git clone https://github.com/cloudposse/terraform-aws-ssm-parameter-store`
1. `make`
1. `make terraform/install TERRAFORM_VERSION=0.12.24`
1. `make terraform/validate`
1. Notice that Terraform validate is not run and the target prints "Terraform 0.12 does not support validate without skipping variables". 

## Environment (please complete the following information):
  - Build Harness Version: master
  - OS: MacOSX
  - Version: 10.15.4

## Additional Context
  - When I run `tf validate` locally without the Makefile / Build-harness -- I get the expected, useful output: 
```
Error: Missing required argument

The argument "region" is required, but was not set.
```
image

### PR DESCRIPTION
**Pull request by Gowiem**
## what
* Updates the `terraform/validate` target to actually run `terraform validate` instead of logging "unsupported" message.

## why
* This is to provide actual validation on v0.12
* Seems the workaround was not known when the "unsupported" log was in place. Now we know we can get around the 'validate' region requirement by supplying `AWS_DEFAULT_REGION`. 

## references
* https://github.com/hashicorp/terraform/issues/21408#issuecomment-495746582

### CODE REVIEW

File: modules/terraform/Makefile [hunk 1]
```diff
@@ -23,9 +23,12 @@ terraform/get-modules:
 	@$(TERRAFORM) init -get -backend=false -input=false >/dev/null
 
 ## Basic terraform sanity check
+## For v0.12.x, we supply a random region to AWS_DEFAULT_REGION so as to fix the
+## "The argument "region" is required" error when validating independent modules.
+## More info: https://github.com/hashicorp/terraform/issues/21408#issuecomment-495746582
 terraform/validate:
```
Comment (osterman) [inline modules/terraform/Makefile:30]: Consider using `make` env localized to the target. 

```suggestion
terraform/validate: AWS_DEFAULT_REGION=us-west-2
terraform/validate:
```

File: modules/terraform/Makefile [hunk 2]
```diff
@@ -23,9 +23,12 @@ terraform/get-modules:
...
 ifeq ("12","$(word 2, $(subst ., ,$(TERRAFORM_VERSION)))")
-	@echo "Terraform 0.12 does not support validate without skipping variables"
+	@AWS_DEFAULT_REGION="us-west-2" $(TERRAFORM) validate
```
Comment (osterman) [inline modules/terraform/Makefile]: See above comment re: env
```suggestion
	@$(TERRAFORM) validate
```

**Comment by Gowiem** [comment]
@osterman ping on this one.

**Comment by Gowiem** [comment]
@osterman any interest in this? Totally fine to close out if not needed. It's been a while and I personally don't need it any longer.

**Comment by Gowiem** [comment]
Closing out due to age and not being needed.

**Pull request closed without merging**